x-pack/legacy/plugins/infra/public/components/logging/log_text_stream/loading_item_view.tsx

```diff
@@ -6,20 +6,36 @@
 
 /* eslint-disable max-classes-per-file */
 
-import { EuiButtonEmpty, EuiIcon, EuiProgress, EuiText } from '@elastic/eui';
-import { FormattedMessage, FormattedRelative } from '@kbn/i18n/react';
+import {
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiLoadingSpinner,
+  EuiButton,
+} from '@elastic/eui';
+import { FormattedMessage, FormattedTime } from '@kbn/i18n/react';
 import * as React from 'react';
+import { Unit } from '@elastic/datemath';
 
 import euiStyled from '../../../../../../common/eui_styled_components';
+import { LogTextSeparator } from './log_text_separator';
+import { extendDatemath } from '../../../utils/datemath';
+
+type Position = 'start' | 'end';
 
 interface LogTextStreamLoadingItemViewProps {
-  alignment: 'top' | 'bottom';
+  position: Position;
+  /** topCursor.time || bottomCursor.time */
+  timestamp?: number;
+  /** startDate || endDate */
+  rangeEdge?: string;
```

Comment:
Noticed this is possibly `'now'` in a later `if` check. What are the other possibilities? Can we be more specific than `string`?

Reply:
The `rangeEdge` is any valid datemath string